### ISSUE TITLE
[IOS] Image.toByteData() is still super slow on Flutter 3.3.7

### ISSUE THREAD

**Issue by fullflash** [flutter/flutter#114788]
`await image.toByteData(format: ImageByteFormat.png);`

After Flutter 3 update it becames SUPER slow on ios. on android it works super fast
on each flutter new version it gets worst.

Flutter 3.3.7 android 7000 ms

Flutter 3 19000ms
Flutter 3.3.7 29000ms

```
Flutter (Channel stable, 3.3.7, on macOS 12.6.1 21G217 darwin-x64, locale
    en-TR)
    • Flutter version 3.3.7 on channel stable at /Users/****/SDK_all/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision e99c9c7cd9 (5 days ago), 2022-11-01 16:59:00 -0700
    • Engine revision 857bd6b74c
    • Dart version 2.18.4
    • DevTools version 2.15.0

[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14B47b
    • CocoaPods version 1.11.2
```

there was an issue that was closed but no fixes still.
https://github.com/flutter/flutter/issues/106773

**Comment by darshankawar** [comment]
@fullflash 
Can you provide [complete minimal code sample](https://stackoverflow.com/help/minimal-reproducible-example) that show the reported behavior ?

**Comment by esDotDev** [comment]
> @fullflash Can you provide [complete minimal code sample](https://stackoverflow.com/help/minimal-reproducible-example) that show the reported behavior ?

An example was already shown in the previous issues that seems to have been incorrectly closed. It has significant details like benchmarks times, and additionak discussion. 

Maybe consider re-opening it? It's quite frustrating to developers when they take the time to log proper details about a bug, it is closed prematurely, then they are asked to provide details again about the same issue:
https://github.com/flutter/flutter/issues/106773

```
import 'dart:ui' as ui;
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';
import 'package:flutter/services.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);
  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  final GlobalKey sharePageKey = GlobalKey();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: RepaintBoundary(
          key: sharePageKey,
          child: Column(
            children: [
              Text(List.generate(200, (index) => "Text").join("#")),
              Image.network(
                "http://pic1.win4000.com/wallpaper/2020-10-10/5f811214aa09e.jpg",
                width: double.infinity,
              ),
            ],
          ),
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () {
          createImage();
        },
        child: const Text("create"),
      ),
    );
  }

  createImage() async {
    final stopWatch = Stopwatch();
    stopWatch.start();
    debugPrint("tim0 ${DateTime.now()}");
    BuildContext? buildContext = sharePageKey.currentContext;
    if (null != buildContext) {
      RenderRepaintBoundary boundary =
          buildContext.findRenderObject() as RenderRepaintBoundary;
      ui.Image image = await boundary.toImage(pixelRatio: 5);
      ByteData? byteData =
          await image.toByteData(format: ui.ImageByteFormat.png);
      if (byteData != null) {
        var pngBytes = byteData.buffer.asUint8List();
        debugPrint("time elapsed in (ms) = ${stopWatch.elapsedMilliseconds}");
      }
      stopWatch.stop();
    }
  }
}
```

**Comment by darshankawar** [comment]
Thanks for the feedback @esDotDev. Using code sample provided above, I do see difference in stable version as compared to master, but it is not the same / close to what OP mentioned:

```
Flutter 3 19000ms
Flutter 3.3.7 29000ms
```

Below are the numbers I got running on iOS simulator:

1. Master (3.5.0-12.0.pre.175):

```
Syncing files to device iPhone 13 Pro...
flutter: tim0 2022-11-09 10:57:58.219414
flutter: time elapsed in (ms) = 676
flutter: tim0 2022-11-09 10:58:03.175285
flutter: time elapsed in (ms) = 1418
flutter: tim0 2022-11-09 10:58:15.377305
flutter: time elapsed in (ms) = 1277
flutter: tim0 2022-11-09 10:58:36.643482
flutter: time elapsed in (ms) = 1164
flutter: tim0 2022-11-09 10:58:41.807146
flutter: time elapsed in (ms) = 1145
flutter: tim0 2022-11-09 10:58:45.692196
flutter: time elapsed in (ms) = 1138
```

2. Stable (3.3.7):

```
Syncing files to device iPhone 13 Pro...
flutter: tim0 2022-11-09 11:20:20.674216
flutter: tim0 2022-11-09 11:20:25.107674
flutter: time elapsed in (ms) = 4648
flutter: tim0 2022-11-09 11:20:28.141560
flutter: time elapsed in (ms) = 4828
flutter: tim0 2022-11-09 11:20:31.784351
flutter: time elapsed in (ms) = 6339
flutter: tim0 2022-11-09 11:20:36.499090
flutter: time elapsed in (ms) = 7241
flutter: tim0 2022-11-09 11:20:42.030696
flutter: time elapsed in (ms) = 7112
flutter: tim0 2022-11-09 11:20:46.117686
flutter: time elapsed in (ms) = 6113
flutter: time elapsed in (ms) = 6584
```

Below are the numbers from stable version running on physical device (iphone 6s) which as compared to above are significantly less.

```
flutter: tim0 2022-11-09 11:31:15.209785
flutter: time elapsed in (ms) = 836
flutter: tim0 2022-11-09 11:31:17.967368
flutter: time elapsed in (ms) = 800
flutter: tim0 2022-11-09 11:31:20.554020
flutter: time elapsed in (ms) = 672
flutter: tim0 2022-11-09 11:31:23.108292
flutter: time elapsed in (ms) = 688
flutter: tim0 2022-11-09 11:31:25.576813
flutter: time elapsed in (ms) = 912
flutter: tim0 2022-11-09 11:31:29.470867
flutter: time elapsed in (ms) = 705
flutter: tim0 2022-11-09 11:31:33.439691
flutter: time elapsed in (ms) = 685
```

Can you share the variance you are seeing ? Based on that we can decide to reopen the original issue.

**Comment by ssyzh** [comment]
https://github.com/flutter/engine/pull/34568,This solved the problem, but the stable branch was not merged, so I had to spend four months at 2.10.5.

**Comment by ssyzh** [comment]
> [flutter/engine#34568](https://github.com/flutter/engine/pull/34568) solved the issue, but the stable branch was not merged, so I had to spend four months at 2.10.5.

Very strange. In 3.3.7, the pull request is not merged, but the speed has been improved.
